src/index/blockfilterindex.cpp

```diff
@@ -146,18 +146,19 @@ bool BlockFilterIndex::CustomCommit(CDBBatch& batch)
     return true;
 }
 
-bool BlockFilterIndex::ReadFilterFromDisk(const FlatFilePos& pos, const uint256& hash, BlockFilter& filter) const
+bool BlockFilterIndex::ReadFilterFromDisk(const FlatFilePos& pos, const uint256& hash, const uint256& block_hash, BlockFilter& filter) const
```

Comment:
nit:
Better if we call it `expected_block_hash`, and also rename the one called `hash` to `expected_filter_hash`.